### PR TITLE
Fix some state panics (hopefully)

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Some panics were observed on a heavily loaded system. Examining the code appeared to show a couple of places where the underlying database connection could be closed before the code that was using it. These holes have been optimistically plugged as it's hard to reproduce the issue.

As a driveby, remove some unused code. And ensure errors are checked when starting state workers.

## QA steps

bootstrap, start the gui, deploy some charms, observed the logs, ensure gui works
destroy a model, ensure that completes

## Bug reference

https://bugs.launchpad.net/bugs/1803629

### CODE REVIEW

**Comment by wallyworld** [comment]
[controller panics](https://trello.com/c/udICoK6v/544-controller-panics)

**Comment by wallyworld** [comment]
!!build!!

**Comment by wallyworld** [comment]
!!build!!

**Comment by wallyworld** [comment]
!!build!!

File: cmd/juju/application/bundle_test.go [hunk 1]
```diff
@@ -731,16 +729,6 @@ func (s *BundleDeployCharmStoreSuite) TestDeployBundleWatcherTimeout(c *gc.C) {
 	// Inject an "AllWatcher" that never delivers a result.
 	ch := make(chan struct{})
 	defer close(ch)
-	watcher := mockAllWatcher{
```
Comment (howbazaar) [inline cmd/juju/application/bundle_test.go]: Was this used for synchronisation? What uses the ch channel now?

File: state/allwatcher.go [hunk 2]
```diff
@@ -1329,10 +1333,14 @@ func (b *allWatcherStateBacking) Changed(all *multiwatcherStore, change watcher.
 
 // Release implements the Backing interface.
 func (b *allWatcherStateBacking) Release() error {
-	// allWatcherStateBacking doesn't need to release anything.
-	return nil
+	// The allWatcher gets a new copy of state so needs to close it.
```
Comment (howbazaar) [inline state/allwatcher.go]: Where is the copy created?

File: state/workers.go [hunk 3]
```diff
@@ -159,9 +170,29 @@ func (ws *workers) allManager(params WatchParams) *storeManager {
...
-	ws.StartWorker(allManagerWorker, func() (worker.Worker, error) {
-		return newStoreManager(newAllWatcherStateBacking(ws.state, params)), nil
-	})
+	if err := ws.StartWorker(allManagerWorker, func() (worker.Worker, error) {
+		// The allWatcher uses system state to load all the entities to watch.
+		// The system state is not ref counted like other states that are fetched
+		// from the state pool, so create a copy here and use that to guard against
+		// the possibility that the system state may be closed elsewhere.
+		// The state copy is closed then the store manager is released.
+		stCopy, err := Open(OpenParams{
```
Comment (howbazaar) [inline state/workers.go]: I think this is the wrong approach. Firstly, using Open causes a new txn watcher to be created for that state object.

We should be more careful about the lifetime of the workers that have dependent objects.

**Pull request closed without merging**